src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -1291,6 +1299,15 @@ private ImmutableArray<BoundNode> GuardedGetBoundNodesFromMap(CSharpSyntaxNode n
             return _guardedNodeMap.TryGetValue(node, out result) ? result : default(ImmutableArray<BoundNode>);
         }
 
+        /// <summary>
+        /// Internal for test purposes only
+        /// </summary>
+        internal ImmutableArray<BoundNode> TestOnlyTryGetBoundNodesFromMap(CSharpSyntaxNode node)
```

Comment:
I've seen some tests using `memberModel.GetUpperBoundNode` (tests on foreach that want to check `EnumeratorInfoOpt`). Could that be used here as well since your validation scenario only check for `IsDefaultOrEmpty`?

Reply:
> I've seen some tests using memberModel.GetUpperBoundNode (tests on foreach that want to check EnumeratorInfoOpt). Could that be used here as well since your validation scenarios only check for IsDefaultOrEmpty?

Those tests just need a node, my tests ensure there is a specific key in the cache, these are different things. GetUpperBoundNode will bind nodes if they are not in the map, I do not want this behavior for my tests.

---
In reply to: [170101953](https://github.com/dotnet/roslyn/pull/24918#discussion_r170101953) [](ancestors = 170101953)